zk.js/src/types/transaction.ts

```diff
@@ -113,7 +114,7 @@ export type ParsedIndexedTransaction = {
 
 // Rpc internal type
 export type RpcIndexedTransaction = {
-  transaction: ParsedIndexedTransaction;
+  transaction: ParsedIndexedTransaction | PublicTransactionIndexerEventBeet;
```

Comment:
i think this causes a lot of the rpc type issues; let's think about how to fix this at the root